PackageExplorer/PackageViewer.xaml.cs

```diff
@@ -489,13 +470,109 @@ private void PackageFolderContextMenu_Opened(object sender, RoutedEventArgs e)
+            if (sender is TreeView treeView && treeView.SelectedItem is PackagePart packagePart)
+            {
+                var data = CreateDataObject(packagePart);
+                Clipboard.SetDataObject(data);
+            }
+        }
+
+        private void OnTreeViewItemPaste(object sender, ExecutedRoutedEventArgs e)
+        {
+            PackageFolder folder = null;
+
+            if (sender is TreeView treeView)
+            {
+                folder = treeView.SelectedItem as PackageFolder;
+            }
+
+            if (HandleDataObject(folder, Clipboard.GetDataObject(), true))
+            {
+                e.Handled = true;
+            }
+        }
+
+        private void OnDataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            if (Clipboard.ContainsData(PackageFileDataFormat))
+            {
+                Clipboard.Clear();
+            }
+        }
+
+        private bool HandleDataObject(PackageFolder folder, IDataObject data, bool copy)
+        {
+            if (DataContext is PackageViewModel model)
+            {
+                if (model.IsSigned || model.IsInEditFileMode || model.IsInEditMetadataMode)
+                {
+                    return false;
+                }
+            }
+
+            if (data.GetDataPresent(PackageFileDataFormat))
+            {
+                if (data.GetData(PackageFileDataFormat) is string packagePartPath)
+                {
+                    folder = folder ?? RootFolder;
+
+                    var packagePart = RootFolder.GetPackageParts().FirstOrDefault(part => part.Path == packagePartPath);
+
+                    if (packagePart != null)
+                    {
+                        if (packagePart is PackageFile file)
+                        {
+                            folder.AddFile(file, copy);
+                        }
+                        else
+                        {
+                            if (packagePart is PackageFolder childFolder && !folder.IsDescendantOf(childFolder))
+                            {
+                                folder.AddFolder(childFolder, copy);
+                            }
+                        }
+                    }
+
+                    return true;
+                }
+            }
+            else if (data.GetDataPresent(DataFormats.FileDrop))
+            {
+                var value = data.GetData(DataFormats.FileDrop);
+                if (value is string[] filenames && filenames.Length > 0)
+                {
+                    var viewModel = DataContext as PackageViewModel;
+                    viewModel.AddDraggedAndDroppedFiles(folder, filenames);
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private IDataObject CreateDataObject(PackagePart packagePart)
+        {
+            var data = new DataObject();
+            data.SetData(PackageFileDataFormat, packagePart.Path);
```

Comment:
During drag & drop the object does not have to be `Serializable` but for copy & past it is required.
That's why now the `PackageFileDataFormat` is just the path and not the actual `PackagePart`.
Since drag & drop from one NPE instance into another one didn't work before this is not a breaking change.